### ISSUE TITLE
Header Label improvements for Screen Reader support 

### ISSUE THREAD

**Issue by saad-mo** [SAP/spartacus#11193]
- This ticket contains improvements for [Headings and Labels Success Criterion](https://github.com/SAP/spartacus/issues/11192)  ([WCAG 2.1 2.4.6](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html))
- master ticket: #11192

### Headings Improvements 
-  [x] Remove h5 from top Nav links and improve markup. Currently, renders `<nav>` for each navigation node. 

### Label Improvements 
- [x] Search Button is identified as an empty group because it is rendered as an icon element, it should be wrapped in a Button. The same applies to the clear button which is visible once when the search box is not empty.
- [x] My Account is H5, should be a link. My Account menu renders several `<nav>` elements, should be `ul>li` structure with only one `<nav>` tag 
- [x] Cart button is identified as a link, should be a button (no change needed)

### Update / Fix things that break
- [x] Update Markup 
- [x] Update SCSS
- [x] Fix / Update unit tests 
- [x] Fix / Update e2e tests 

![header-labels](https://user-images.githubusercontent.com/15215400/108417021-15b45c80-71f5-11eb-9802-7830293930c4.png)
